rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/custom_participant_info.hpp

```diff
@@ -37,6 +37,7 @@ typedef struct CustomParticipantInfo
   ReaderInfo * secondarySubListener;
   WriterInfo * secondaryPubListener;
   rmw_guard_condition_t * graph_guard_condition;
+  bool leave_middleware_default_qos;
```

Comment:
I think it would be helpful here to have a bit of documentation describing the behavior of each state.

Reply:
I agree. I have added an explanation for this attribute because only with the name it could generate misunderstandings